src/Interpreters/Cache/LRUFileCachePriority.cpp

```diff
@@ -63,12 +64,13 @@ IFileCachePriority::Iterator LRUFileCachePriority::add(
 void LRUFileCachePriority::removeAll(const CacheGuard::Lock &)
 {
     CurrentMetrics::sub(CurrentMetrics::FilesystemCacheSize, current_size);
-    CurrentMetrics::sub(CurrentMetrics::FilesystemCacheElements, queue.size());
+    CurrentMetrics::sub(CurrentMetrics::FilesystemCacheElements, current_elements_num);
 
     LOG_TEST(log, "Removed all entries from LRU queue");
 
     queue.clear();
     current_size = 0;
+    current_elements_num = 0;
```

Comment:
Looks strange. Shouldn't we set metrics after zeroing size and number of elements?